### PR TITLE
Refactored Advanced PBR Terrain to use new for-loops

### PR DESCRIPTION
**Pull request by yaRnMcDonuts**
Similar to my recent pull request doing the same for the base PbrTerrain shader (https://github.com/jMonkeyEngine/jmonkeyengine/pull/1901) this PR also adds for-loop support to the advanced version of the pbr shader that uses texture arrays

### CODE REVIEW

File: jme3-terrain/src/main/resources/Common/MatDefs/Terrain/AdvancedPBRTerrain.frag [hunk 1]
```diff
@@ -2,83 +2,73 @@
 #import "Common/ShaderLib/GLSLCompat.glsllib"
 #import "Common/ShaderLib/PBR.glsllib"
 #import "Common/ShaderLib/Lighting.glsllib"
-#import "Common/MatDefs/Terrain/AfflictionLib.glsllib"
-
+#import "MatDefs/ShaderLib/AfflictionLib.glsllib"
```
Comment (Ali-RS) [inline jme3-terrain/src/main/resources/Common/MatDefs/Terrain/AdvancedPBRTerrain.frag:5]: `#import "MatDefs/ShaderLib/AfflictionLib.glsllib"`

Can you please recheck this path?

Looks like the library is in a different path:

https://github.com/jMonkeyEngine/jmonkeyengine/tree/a04eba4c413c1e1417a204b693d53b340441e701/jme3-terrain/src/main/resources/Common/MatDefs/Terrain

**Comment by Ali-RS** [comment]
Have you tested this to see if the `PBRTerrainAdvancedTest` running without an issue?

**Comment by yaRnMcDonuts** [comment]
I did, but I tested it with an updated copy of that glsllib in the test cases assets directory, and I forgot change the path to point back to the proper Common/MatDefs folder where its located in jme's terrain library. 

I just updated this PR so it should use the proper path for both terrain shaders now, sorry for the mistake.

**Comment by Ali-RS** [comment]
No worries, Thanks!

**Comment by Ali-RS** [comment]
I want to merge this PR and https://github.com/jMonkeyEngine/jmonkeyengine/pull/1905 in about 12H and then release alpha2.

**Pull request merged**